rocAL/rocAL_pybind/rocal_pybind.cpp

```diff
@@ -316,6 +332,162 @@ namespace rocal{
+                return rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of tensor.
+                )code"
+            )
+            .def("torch_dtype", [](rocalTensor &output_tensor) {
+                return "torch." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of torch tensor.
+                )code"
+            )
+            .def("numpy_dtype", [](rocalTensor &output_tensor) {
+                return "np." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of numpy array.
+                )code"
+            )
+            .def("cupy_dtype", [](rocalTensor &output_tensor) {
+                return "cp." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of cupy array.
+                )code"
+            )
+            .def("dims", [](rocalTensor &output_tensor) {
+                return output_tensor.info().dims();
+            },
+                R"code(
+                Returns dims of tensor.
+                )code"
+            )
+            m.def("dimensions", []=(rocalTensor &output_tensor) {
+            py::list list;
+            for (uint i = 0; i < output_tensor.info().dims(); i++)
+                list.append(output_tensor.info().dims().at(i));
+            return list; 
+            });
+            .def(
+            "copy_data", [](rocalTensor &output_tensor, py::object p) {
+                auto ptr = ctypes_void_ptr(p);
+                output_tensor.copy_data(ptr);
+            },
+                R"code(
+                Copies the ring buffer data to python buffer pointers.
+                )code"
+            )
+            .def(
+                "at",
+                [](rocalTensor &output_tensor, uint idx) {
+                    const auto& info = output_tensor.info();
+                    uint idx_stride = 1;
+                    for (uint i = 0; i < info.num_of_dims() - 1 ; i++) idx_stride*=output_tensor.info().dims().at(i+1);
+                    std::vector<size_t> stride_per_sample(info.strides());
+                    stride_per_sample.erase(stride_per_sample.begin());
+                    std::vector<size_t> dims(info.dims());
+                    dims.erase(dims.begin());
+                    
+                    switch(output_tensor.info().data_type()) 
+                    {
+                        case RocalTensorDataType::UINT8:
+
+                            return py::array(py::buffer_info(
+                                ((unsigned char *)(output_tensor.buffer())) + idx * idx_stride,
+                                sizeof(unsigned char),
+                                py::format_descriptor<unsigned char>::format(),
```

Comment:
In future we can have a way to translate the size of datatype also from the datatype using the map or something

Reply:
We can, but we will need changes for the format descriptor too